feat: add `usePwsh` helper for PowerShell v7+

ShGKme opened the pull request:
Fixes https://github.com/google/zx/issues/785

- [x] Tests pass
- [x] Appropriate changes to README are included in PR
- [x] Types updated

## Notes

No documentation was added. I haven't found any documentation for other helpers apart from release notes...

A smoke test was added to `win32` test but I'm not sure if it's a good idea.
- `pwsh` is not available by default
- `pwsh` can be used on any supported platform

So the last commit can be dropped. Or I can create a new smoke test.

## Description

A new helper `usePwsh` to use `pwsh` shell (PowerShell v7+).

```js
import { usePowerShell, usePwsh } from 'zx'

usePowerShell() // to enable powershell.exe (PowerShell v5 on Windows)
usePwsh()       // switch to pwsh (PowerShell v7 on any supported platform)
```

## Usage

Tested on Windows 11 and Fedora 38

<table>

<tr><td>

```js
usePwsh()
console.log((await $`echo 'Foo' && echo 'Bar'`).stdout)
```

</td><td>

```
Foo
Bar
```

</td>

</tr><tr><td>

```js
usePowerShell()
console.log((await $`echo 'Foo' && echo 'Bar'`).stdout)
```

</td><td>

```
At line:1 char:12
+ echo 'Foo' && echo 'Bar'; exit $LastExitCode
+            ~~
The token '&&' is not a valid statement separator in this version.
    + CategoryInfo          : ParserError: (:) [], ParentContainsErrorRecordException
    + FullyQualifiedErrorId : InvalidEndOfLine
```

</td></tr>

antongolub commented on test/smoke/win32.test.js:
```diff
@@ -38,4 +41,13 @@ _describe('win32', () => {
       assert.match(p.stdout, /Windows 'rulez!'/)
     })
   })
+
```
let's _keep_ cov treshold. Add plz smth like core.test.js:

```ts
test('usePwsh() sets proper defaults', () => {
   const wsync = which.sync
   which.sync = bin => bin === 'pwsh' ? 'pwsh' : wsync(bin)
   
   usePwsh()
   assert.equal($.shell, 'pwsh')
   assert.equal($.prefix, '...')
   // ...
   which.sync = wsync
})
```

ShGKme replied:
Instead of this win32 smoke test, right?

antongolub replied:
as addition

antongolub commented on test/core.test.js:
```diff
@@ -583,4 +583,15 @@ describe('core', () => {
+    assert.equal($.prefix, '')
+    assert.equal($.postfix, '; exit $LastExitCode')
+    assert.equal($.quote, quotePowerShell)
+    which.sync = originalWhichSync
```
+useBash() at the end to revert $ presets

ShGKme commented:
Fixed smoke test to use `which` from `build/vendors`, made `Test / smoke-win32-node16 (pull_request)` green 🟢

Added `useBash` and removed unneeded import from unit tests (having import with globals).

Fixed formatting.

antongolub commented on test/smoke/win32.test.js:
```diff
@@ -15,9 +15,12 @@
 import assert from 'node:assert'
 import { test, describe } from 'node:test'
 import '../../build/globals.js'
+import { which } from '../../build/vendor.js'
```
`import '../../build/globals.js'` goes upper, so `which` API should be available without extra import.

antongolub reviewed (approved):
lgtm

antongolub commented:
@ShGKme, thanks for the improvement.

Could you also verify the snapshot zx@8.0.2-dev.bc2a08c?

ShGKme commented:
> Could you also verify the snapshot [zx@8.0.2-dev.bc2a08c](https://www.npmjs.com/package/zx/v/8.0.2-dev.bc2a08c)?

✅ Verified  

- Tested in https://github.com/nextcloud/talk-desktop/pull/642, works fine on Windows.
- Tested on Fedora 38
  ![image](https://github.com/google/zx/assets/25978914/45e7a531-1b0e-4ddf-9962-7f1cd0a1e06d)
